backend/src/main/java/com/trackpack/app/keys/UserParcelKey.java

```diff
@@ -0,0 +1,31 @@
+package com.trackpack.app.keys;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import javax.persistence.Embeddable;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.UUID;
+
+@Embeddable
+@NoArgsConstructor
+@Data
+public class UserParcelKey implements Serializable {
```

Comment:
I think its better to remove Key entity and just add generated ID to UserParcel.